dlc-node-test/src/test/scala/org/bitcoins/dlc/node/DLCServerTorTest.scala

```diff
@@ -0,0 +1,163 @@
+              serverConnectionHandlerOpt = Some(connectionHandler)
+              serverProbe.ref
+            }
+          ))
+
+        val resultF: Future[Future[Assertion]] = for {
+          _ <- boundAddressPromise.future
+          connectedAddressPromise = Promise[InetSocketAddress]()
+          connectAddress <- connectAddressF
+        } yield {
+          var clientConnectionHandlerOpt = Option.empty[ActorRef]
+          val clientProbe = TestProbe()
+          val client = TestActorRef(
+            DLCClient.props(
+              fundedDLCWallet.wallet,
+              Some(connectedAddressPromise),
+              None,
+              { (_, _, connectionHandler) =>
+                clientConnectionHandlerOpt = Some(connectionHandler)
+                clientProbe.ref
+              }
+            ))
+
+          client ! DLCClient.Connect(
+            Peer(connectAddress,
+                 socks5ProxyParams = Some(
+                   Socks5ProxyParams(
+                     address = torProxyAddress,
+                     credentialsOpt = None,
+                     randomizeCredentials = true
+                   ))))
+
+          for {
+            _ <- connectedAddressPromise.future
+            _ <- AsyncUtil.retryUntilSatisfied(
+              serverConnectionHandlerOpt.isDefined)
+            _ <- AsyncUtil.retryUntilSatisfied(
+              clientConnectionHandlerOpt.isDefined)
+            pingTLV =
+              PingTLV(
+                UInt16.one,
+                ByteVector.fromValidHex("00112233445566778899aabbccddeeff"))
+            clientConnectionHandler = clientConnectionHandlerOpt.get
+            _ = clientProbe.send(clientConnectionHandler, pingTLV)
+            _ = serverProbe.expectMsg(timeout, LnMessage(pingTLV))
+            pongTLV = PongTLV.forIgnored(
+              ByteVector.fromValidHex("00112233445566778899aabbccddeeff"))
+            serverConnectionHandler = serverConnectionHandlerOpt.get
+            _ = serverProbe.send(serverConnectionHandler, pongTLV)
+            _ = clientProbe.expectMsg(timeout, LnMessage(pongTLV))
+            // 131063 - is a magic size for OS X when this test case starts failing (131073 overall TLV size)
+            ignored = ByteVector.fill(65000)(0x55)
+            bigTLV = PongTLV.forIgnored(ignored)
+            _ = clientProbe.send(clientConnectionHandler, bigTLV)
+            _ = serverProbe.expectMsg(timeout, LnMessage(bigTLV))
+            _ = clientProbe.send(clientConnectionHandler,
+                                 DLCConnectionHandler.CloseConnection)
+            _ = clientProbe.send(clientConnectionHandler, pingTLV)
+            _ = serverProbe.expectNoMessage()
+            _ = serverProbe.send(serverConnectionHandler, pongTLV)
+            _ = clientProbe.expectNoMessage()
+          } yield {
+            succeed
+          }
+        }
+        resultF.flatten
+    }
+
+    resultF
+  }
+
+  private def portIsBound(address: InetSocketAddress): Boolean =
```

Comment:
nit: This seems like a really useful utility, i'm surprised we don't have it anywhere else yet. Any thoughts on where to put it so we don't have to re-implement everytime?